### DISCUSSION TITLE
zeus preview for v1 GTR 3 app error

### DISCUSSION THREAD

**Discussion by 0ffTopic** [zepp-health/zeppos-forum#331] [Q&A] [answered]
Hello

I mande a simple app for GTR3 using the Guides for ZEPP OS v1, I am able to preview in the Simulator using zeus dev command, but when I try to build a preview package I get the error:

PS ...> zeus preview
info  building...
info  Start building package, targets: gtr-3. 
[ROLLUP] Transform 4 JS files in 1365ms 
[RESIZE] Succeed resize icon.png to target size 86, took 22ms 
[PNG2TGA] Converting PNG files... 
[PNG2TGA] 1 files, 83ms, done! 
[QJSC] Compiling JS files... 
[QJSC] 2 files, 66ms, done! 
undefined 
error  TypeError: Cannot read properties of undefined (reading 'replace') 
    at ..\npm\node_modules\@zeppos\zeus-cli\private-modules\zeppos-app-utils\dist\modules\build.js:156:44 
    at step (..\npm\node_modules\@zeppos\zeus-cli\private-modules\zeppos-app-utils\dist\modules\build.js:56:23) 
    at Object.next (..\npm\node_modules\@zeppos\zeus-cli\private-modules\zeppos-app-utils\dist\modules\build.js:37:53) 
    at fulfilled (..\npm\node_modules\@zeppos\zeus-cli\private-modules\zeppos-app-utils\dist\modules\build.js:28:58) 
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) 
info  getting package... 
error  getting package failed.

Anyone has ever encoutered this issue?

**Comment by silver-zepp** [comment]
Hey, please check that your nodejs/npm and zeus-cli and Simulator versions are the most recent, update if they are not.
If that won't help you can share your project so we can look into it. Also, you can try to use the older version of GTR3 emulator and see if that helps.

**Reply by 0ffTopic** [reply]
Node.js - 20.12.2.
Zeus - 1.5.11 (zpm v3.0.21)
Simulator - 1.2.4

As I was able to check these are the latest versions. Just to clarify the problem is not running the app on the emulator, I can preview my app if I run _zeus dev_. The problem occurs only when I try to execute _zeus build_ or _zeus preview_.
May also be relevant, the app was generated with the todo list sample. I've also created a new project only with the base ToDo sample and the code builds. Here is my code:
[tps.zip](https://github.com/zepp-health/zeppos-forum/files/14994418/tps.zip)

**Reply by silver-zepp** [reply, marked as answer]
Thanks for the details and the project attached. The difference between `zeus dev` and other commands is that they are cloud-based, while `dev` runs local.  By the way, do you experience the same behavior running the fetch API example from `zeus create`?

Considering that the issue is cloud-based, that hints that the problem might be that your app is failing one of our server checks, and if I had to guess it would be the app name "Tenis/Padel Score". Try removing the forward slash "/" from it and see if it works. If that's the case, you can use "-" instead.

**Reply by 0ffTopic** [reply]
That was the problem. Replacing the / by the - solved my build issue.

Thanks
